### PR TITLE
fix(platform): fix fdp-menu-item extra spaces

### PR DESCRIPTION
**Pull request by dimamarksman**
#### Please provide a link to the associated issue.
Closes: https://github.com/SAP/fundamental-ngx/issues/5461

#### Please provide a brief summary of this pull request.
This fixes platform menu-item structure to follow fundamental-styles
#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- [na] update `README.md`
- [na] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [na] Documentation Examples
- [na] Stackblitz works for all examples

### CODE REVIEW

File: libs/platform/src/lib/components/menu/menu-item.component.ts [hunk 1]
```diff
@@ -8,73 +8,105 @@ import {
...
+    isTrigger = false;
+
+    /** @hidden */
+    _isSelected = false;
+    /**
+     * @hidden
+     * set CSS class 'is-selected' if menu-item opens a sub-menu.
+     */
+    set isSelected(selected: boolean) {
+        const changed = selected !== this._isSelected;
```
Comment (N1XUS) [inline libs/platform/src/lib/components/menu/menu-item.component.ts:55]: Why not use comparison directly in condition?

Comment (dimamarksman) [reply libs/platform/src/lib/components/menu/menu-item.component.ts]: moved it to the condition

**Review by N1XUS — APPROVED** [review]

**Comment by InnaAtanasova** [comment]
Please add more meaningful name for the PR

File: libs/platform/src/lib/components/menu/menu-item.component.ts [hunk 2]
```diff
@@ -8,73 +8,104 @@ import {
...
-    encapsulation: ViewEncapsulation.None
+    encapsulation: ViewEncapsulation.None,
+    host: {
+        class: 'fd-menu__item',
+        tabindex: '-1'
+    }
 })
 export class MenuItemComponent implements OnDestroy, FocusableOption {
-    @Input() cascadeDirection: 'right' | 'left' = 'right';
+    @Input()
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/menu/menu-item.component.ts:34]: typedocs

File: libs/platform/src/lib/components/menu/menu.component.ts [hunk 3]
```diff
@@ -153,7 +158,7 @@ export class MenuComponent implements AfterViewInit, AfterContentInit, OnDestroy
         }
     }
 
-    onClick(event: MouseEvent): void {
+    onClick(): void {
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/menu/menu.component.ts:161]: typedocs

**Review by InnaAtanasova — APPROVED** [review]

**Review by platon-rov — APPROVED** [review]

**Comment by InnaAtanasova** [comment]
Please resolve the conflicts. The rest is 🚀

File: libs/platform/src/lib/menu/menu-trigger.directive.ts [hunk 4]
```diff
@@ -59,13 +42,33 @@ export class MenuTriggerDirective implements OnDestroy, AfterContentInit {
...
+    /** @hidden */
     private _isMenuOpen = false;
+    /** @hidden */
+    private outsideClickSubscription: Subscription = Subscription.EMPTY;
+    /** @hidden */
+    private menuCloseSubscription: Subscription = Subscription.EMPTY;
+    /** @hidden */
+    private parentMenuCloseSubscription: Subscription = Subscription.EMPTY;
+    /** @hidden */
+    private menuItemHoverChangeSubscription: Subscription = Subscription.EMPTY;
```
Comment (Betrozov) [inline libs/platform/src/lib/menu/menu-trigger.directive.ts:60]: must be with prefix `_`

**Review by Betrozov — COMMENTED** [review]
please check all the places where you added doctype and add prefix if it needed

File: libs/platform/src/lib/menu/menu-trigger.directive.ts (hunk 4 above)
Comment (dimamarksman) [reply libs/platform/src/lib/menu/menu-trigger.directive.ts]: prefixed

**Pull request merged**